common/state-machine-lib/src/main/java/org/eclipse/dataspaceconnector/common/statemachine/EntitiesProcessor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.common.statemachine;
+
+/**
+ * Interface that declares an abstraction for a component that process some entities and return the number of the processed ones.
+ * Used by {@link StateMachineLoop} to decide whether to apply wait strategy in loop iteration
+ *
+ */
+public interface EntitiesProcessor {
```

Comment:
I also wonder if we should rename this now to something more generic to reflect that it is a processing stage or step?

Reply:
At first I called it `Processor` but it seemed too generic, I agree on look for a better name for it but at the moment I don't have better ideas.

Reply:
@paullatzelsperger do you have an idea for a name? `StateMachineProcessor`?

Reply:
those are champagne problems to have :D

Reply:
@ndr-brt looks good, can you just ping me when you are ready with this?